src/frontend/src/lib/components/registry/players/PlayerRegistrationHistory.svelte

```diff
@@ -0,0 +1,66 @@
+<script lang="ts">
+  import { page } from '$app/stores';
+  import { user } from '$lib/stores/stores';
+  import type { PlayerInfo } from '$lib/types/player-info';
+  import type { UserInfo } from '$lib/types/user-info';
+  import type { PlayerTransferItem } from '$lib/types/player-transfer';
+  import Section from '$lib/components/common/Section.svelte';
+  import Table from '$lib/components/common/Table.svelte';
+  import LL from '$i18n/i18n-svelte';
+  import { onMount } from 'svelte';
+  let history: PlayerTransferItem[] = [];
+  let user_info: UserInfo;
+  user.subscribe((value) => {
+    user_info = value;
+  });
+  export let player: PlayerInfo;
+
+  function toDate(unix_timestamp: number) {
+    return new Date(unix_timestamp * 1000).toLocaleDateString();
+  }
+
+  async function fetchData() {
+    // API
+    let url = `/api/registry/players/${player.id}/getPlayerTransferHistory`;
+    const res = await fetch(url);
+    if (res.status !== 200) {
+      return;
+    }
+    let body = await res.json();
+    history = body.history;
+  }
+
+  onMount(fetchData);
+</script>
+
+<Section header={$LL.REGISTRATION_HISTORY.REGISTRATION_HISTORY()}>
+  <div>
+    <div>
+      <Table>
+        <thead>
+          <tr>
+            <th>Team</th>
+            <th>Registration Period</th>
+          </tr>
+        </thead>
+        <tbody>
+          {#each history as record, i}
+            <tr class="row-{i % 2}">
+              <td>
+                <a
+                  class="hover:text-emerald-400"
+                  href="/{$page.params.lang}/registry/teams/profile?id={record.team_id}"
+                >
+                  {record.roster_name ? record.roster_name : record.team_name}
+                </a>
+              </td>
+              <td>
+                {toDate(record.join_date)} - {record.leave_date ? toDate(record.leave_date) : 'Present'}
+              </td>
+            </tr>
+          {/each}
+        </tbody>
+      </Table>
+    </div>
+  </div></Section
+>
```

Comment:
Going to use this to pick on all 3 of registration history, player tournament history, and team tournament history, but these sections/tables shouldn't be displayed if the length of the API response array is 0, since it just displays table headings with no data which looks pretty ugly otherwise. See this screenshot of a player's team tournament history from my dev environment:
![image](https://github.com/user-attachments/assets/254cbebb-b39c-4fe7-bd9c-b5c305644b34)

Also, like I said when commenting on the API command for getting this data, we should have mandatory game/mode filters here, otherwise the history can quickly be impossible to understand:
![image](https://github.com/user-attachments/assets/228393eb-3a55-460d-950b-3f94d4d69b2d)

Reply:
OK that is doable for sure.
I have some extended vacation time coming up after Thanksgiving. I'll plan to work on this + the API changes for mandatory game+mode during that time.

Thanks for the feedback!